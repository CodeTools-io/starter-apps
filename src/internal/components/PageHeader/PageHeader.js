import React from 'react';
import { Grid, Heading, Paragraph } from 'grommet';
import Socials from 'internal/components/Socials';

export default function PageHeader({
  title,
  description,
  socials = {
    github: true,
    twitter: true,
  },
  ...props
}) {
  return (
    <Grid
      className="PageHeader"
      columns={['1/4', '1/4', '1/4', '1/4']}
      rows={['auto']}
      areas={[
        ['heading', 'heading', 'heading', 'heading'],
        ['intro', 'intro', 'intro', 'intro'],
        ['socials', 'socials', 'socials', 'socials'],
      ]}
      align="center"
      alignContent="center"
      gap="small"
      {...props}
    >
      <Heading gridArea="heading" level={1} margin="none">
        {title}
      </Heading>
      <Paragraph gridArea="intro" margin="none" fill>
        {description}
      </Paragraph>
      <Socials gridArea="socials" margin={{ top: 'small' }} {...socials} />
    </Grid>
  );
}
