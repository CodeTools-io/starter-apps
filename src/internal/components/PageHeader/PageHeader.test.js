import React from 'react'
import PageHeader from './PageHeader'

describe('PageHeader', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <PageHeader>
        <p>Content for the PageHeader</p>
      </PageHeader>
    )

    expect(container).toMatchSnapshot()
  })
})

