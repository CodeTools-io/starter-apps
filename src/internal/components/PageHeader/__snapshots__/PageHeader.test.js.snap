// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader can render 1`] = `
<div>
  <div
    class="StyledGrid-sc-1wofa1l-0 xUOVV PageHeader"
  >
    <h1
      class="StyledHeading-sc-1rdh4aw-0 cxHzQU"
    >
      Example of title
    </h1>
    <p
      class="StyledParagraph-tbetod-0 eYnGVp"
    >
      A short description about the page
    </p>
  </div>
</div>
`;
